test/e2e/major_upgrade_v3.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/v3/app"
+	v2 "github.com/celestiaorg/celestia-app/v3/pkg/appconsts/v2"
+	v3 "github.com/celestiaorg/celestia-app/v3/pkg/appconsts/v3"
+	"github.com/celestiaorg/celestia-app/v3/test/e2e/testnet"
+)
+
+func MajorUpgradeToV3(logger *log.Logger) error {
+	numNodes := 4
+	upgradeHeightV3 := int64(20)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger.Println("Creating testnet")
+	testNet, err := testnet.New(ctx, "MajorUpgradeToV3", seed, nil, "test")
+	testnet.NoError("failed to create testnet", err)
+
+	defer testNet.Cleanup(ctx)
+
+	// HACKHACK: use a version of celestia-app built from a commit on this PR.
+	// This can be removed after the PR is merged to main.
+	version := "pr-3910"
```

Comment:
obligatory github comment to accompany this code comment

Reply:
Thanks! Opened https://github.com/celestiaorg/celestia-app/issues/3947